### PR TITLE
feat(mobile): pwa and browser theme-color optimization

### PR DESCRIPTION
**Pull request by CatsJuice**
[AF-1325](https://linear.app/affine-design/issue/AF-1325/优化-pwa-体验), [AF-1317](https://linear.app/affine-design/issue/AF-1317/优化：-pwa-的顶部-status-bar-颜色应与背景保持一致), [AF-1318](https://linear.app/affine-design/issue/AF-1318/优化：pwa-的底部应当有符合设备安全高度的padding), [AF-1321](https://linear.app/affine-design/issue/AF-1321/更新一下-fail-的-pwa-icon)

- New `<SafeArea />` ui component
- New `useThemeColorV1` / `useThemeColorV2` hook:
    - to modify `<meta name="theme-color" />` with given theme key

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#8168** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8168?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/component/src/hooks/use-theme-color-meta.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  // create and append meta
+  meta = document.createElement('meta');
+  meta.name = 'theme-color';
+  document.head.append(meta);
+  return meta;
+}
+
+export const useThemeColorMeta = (color: string) => {
+  useEffect(() => {
```
Comment (forehalo) [inline packages/frontend/component/src/hooks/use-theme-color-meta.ts]: useLayoutEffect is more suitable

File: packages/frontend/component/src/ui/safe-area/style.css.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export const safeArea = style({
+  selectors: {
+    '&[data-top]': {
+      paddingTop: `calc(${topOffsetVar} + 12px)`,
+    },
+    '&[data-bottom]': {
+      paddingBottom: `calc(${bottomOffsetVar} + 0px)`,
+    },
+    '&[data-standalone][data-top]': {
+      paddingTop: `calc(env(safe-area-inset-top, 12px) + ${topOffsetVar})`,
```
Comment (pengx17) [inline packages/frontend/component/src/ui/safe-area/style.css.ts:15]: do we need to check if `env` feature is available?

File: packages/frontend/component/src/hooks/use-theme-color-meta.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  // create and append meta
+  meta = document.createElement('meta');
+  meta.name = 'theme-color';
+  document.head.append(meta);
+  return meta;
+}
+
+export const useThemeColorMeta = (color: string) => {
```
Comment (pengx17) [inline packages/frontend/component/src/hooks/use-theme-color-meta.ts:23]: what is the use case?

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/component/src/ui/safe-area/style.css.ts (hunk 2 above)
Comment (CatsJuice) [reply packages/frontend/component/src/ui/safe-area/style.css.ts:15]: I think it's stable enough: [https://caniuse.com/?search=env()](https://caniuse.com/?search=env())

File: packages/frontend/component/src/hooks/use-theme-color-meta.ts (hunk 3 above)
Comment (CatsJuice) [reply packages/frontend/component/src/hooks/use-theme-color-meta.ts:23]: Used for `useThemeColorV1()`/ `useThemeColorV2()` to change browser's header color dynamically for different page or different state.

Comment (pengx17) [reply packages/frontend/component/src/hooks/use-theme-color-meta.ts:23]: it seems next-themes already support adding meta tag https://github.com/pacocoursey/next-themes/pull/114

Comment (pengx17) [reply packages/frontend/component/src/hooks/use-theme-color-meta.ts:23]: it seems next-themes already support adding meta tag https://github.com/pacocoursey/next-themes/pull/114
Ops it is only available for beta version of next-themes

Comment (CatsJuice) [reply packages/frontend/component/src/hooks/use-theme-color-meta.ts:23]: We need to change `theme-color` for different routes, and even different state like "when modal is opened"

**Pull request merged**